src/nonces.rs

```diff
@@ -0,0 +1,195 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::frost::keys::SigningShare;
+use crate::frost::round1::SigningNonces;
+use crate::participant::Identity;
+use crate::participant::IdentitySerialization;
+use rand_chacha::ChaCha20Rng;
+use rand_core::SeedableRng;
+use std::borrow::Borrow;
+
+type ParticipantCount = u32;
+
+fn nonces_seed<I>(
+    secret: &SigningShare,
+    transaction_hash: &[u8],
+    signing_participants: &[I],
+) -> <ChaCha20Rng as SeedableRng>::Seed
+where
+    I: Borrow<Identity>,
+{
+    let mut identifiers: Vec<IdentitySerialization> = signing_participants
+        .iter()
+        .map(|id| id.borrow().serialize())
+        .collect();
+    let count: ParticipantCount = identifiers
+        .len()
+        .try_into()
+        .expect("too many signing_participants");
+    // TODO This sorts the identifiers, so that the order of the input does not influence the
+    // output, but does not remove duplicates. Should that case be handled?
```

Comment:
I think it's reasonable to handle duplicates; either remove them or have unique identifiers as a pre-condition.